.github/workflows/build-and-integration-tests.yml

```diff
@@ -87,13 +87,18 @@ jobs:
         override: true
 
     - name: Configure caching
-      uses: actions/cache@v2
+      uses: actions/cache@v3
       with:
-        key: ${{ matrix.os }}-${{ matrix.target }}
         path: |
-          ${{ env.HOME }}/.cargo/git
-          ${{ env.HOME }}/.cargo/registry
-          target
+          ~/.cargo/bin/
```

Comment:
I prefer not to have `~/.cargo/bin` cached since the integration tests installed tools into it.
If it is cached and `cargo-binstall` failed, then the CI would success regardless.